src/mgclient.c

```diff
@@ -470,10 +488,12 @@ static int init_tcp_connection(const mg_session_params *params, int *sockfd,
     return status;
   }
 
+#ifndef __EMSCRIPTEN__
   int set_options_status = mg_socket_options(tsockfd, session);
   if (set_options_status != MG_SUCCESS) {
     return set_options_status;
   }
+#endif
```

Comment:
I think this `ifndef` should be pushed to `mg_socket_options`, because we might want to set different options for emscripten in the future.